src/gui/settingsdialog.cpp

```diff
@@ -242,9 +245,13 @@ SettingsDialog::SettingsPage SettingsDialog::currentPage() const
 void SettingsDialog::setCurrentAccount(Account *account)
 {
     _currentAccount = account;
-    _ui->stack->setCurrentWidget(accountSettings(account));
-    _currentPage = SettingsPage::Account;
-
+    if (account) {
+        _ui->stack->setCurrentWidget(accountSettings(account));
+        _currentPage = SettingsPage::Account;
+    } else {
+        // go to the settings page if the last account was removed
+        _ui->stack->setCurrentWidget(_generalSettings);
```

Comment:
In this case, `_currentPage` is not changed at all. No idea if that is a problem or not...